src/api/ValuesApi.js

```diff
@@ -0,0 +1,114 @@
+import { polyfill } from 'es6-promise';
+import 'isomorphic-fetch';
+
+polyfill();
+
+export default {
+  name: 'ValuesApi',
+
+  getValues(name, params, options, searchState, qtext, facetObject) {
+    //get options
+    return fetch('/v1/config/query/all?format=json', {
+      method: 'GET',
+      headers: {
+        'Content-Type': 'application/json',
+        Accept: 'application/json'
+      },
+      credentials: 'same-origin'
+    }).then(
+      response => {
+        return response.json().then(function(json) {
+          let options = json.options || {};
+
+          let path = '';
+          let collation = '';
+          for (let i=0; i < options.constraint.length; i++) {
+            if (options.constraint[i].name === name){
+              path = options.constraint[i].range['path-index'].text;
+              collation = options.constraint[i].range.collation;
+            }
+          }
+
+          options.values = {
+            name: name,
+            range: {
+              collation: collation,
+              type: 'xs:string',
+              'path-index': {
+                text: path,
+                ns: ''
+              }
+            },
```

Comment:
I think it is better to just copy the entire range object from the constraint. Might not be string, might not be path index..